gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -80,15 +80,163 @@ enum
   PROP_INPUT,
   PROP_INPUTTYPE,
   PROP_INPUTNAME,
+  PROP_INPUTLAYOUT,
   PROP_OUTPUT,
   PROP_OUTPUTTYPE,
   PROP_OUTPUTNAME,
+  PROP_OUTPUTLAYOUT,
   PROP_CUSTOM,
   PROP_SUBPLUGINS,
   PROP_ACCELERATOR,
   PROP_IS_UPDATABLE,
 };
 
+/**
+ * @brief Initialize the tensors layout.
+ */
+static void
+gst_tensors_layout_init (tensors_layout layout)
+{
+  int i;
+
+  for (i = 0; i < NNS_TENSOR_SIZE_LIMIT; i++) {
+    layout[i] = _NNS_LAYOUT_ANY;
+  }
+}
+
+/**
+ * @brief Get tensor layout from string input.
+ * @return Corresponding tensor_layout.
+ */
+static tensor_layout
+gst_tensor_parse_layout_string (const gchar * layoutstr)
+{
+  gsize len;
+  gchar *layout_string;
+  tensor_layout layout = _NNS_LAYOUT_ANY;
+
+  if (layoutstr == NULL)
+    return layout;
+
+  /* remove spaces */
+  layout_string = g_strdup (layoutstr);
+  g_strstrip (layout_string);
+
+  len = strlen (layout_string);
+
+  if (len == 0) {
+    g_free (layout_string);
+    return layout;
+  }
+
+  if (g_ascii_strcasecmp (layoutstr, "NCHW") == 0) {
+    layout = _NNS_LAYOUT_NCHW;
+  } else if (g_ascii_strcasecmp (layoutstr, "NHWC") == 0) {
+    layout = _NNS_LAYOUT_NHWC;
+  } else if (g_ascii_strcasecmp (layoutstr, "ANY") == 0) {
+    layout = _NNS_LAYOUT_ANY;
+  } else {
+    GST_WARNING ("Invalid layout, defaulting to none layout.");
+    layout = _NNS_LAYOUT_NONE;
+  }
+
+  g_free (layout_string);
+  return layout;
+}
+
+/**
+ * @brief Parse the string of tensor layouts
+ * @param layout layout of the tensors
+ * @param layout_string string of layout
+ * @return number of parsed layouts
+ */
+static guint
+gst_tensors_parse_layouts_string (tensors_layout layout,
+    const gchar * layout_string)
+{
+  guint num_layouts = 0;
+
+  g_return_val_if_fail (layout != NULL, 0);
+
+  if (layout_string) {
+    guint i;
+    gchar **str_layouts;
+
+    str_layouts = g_strsplit_set (layout_string, ",.", -1);
+    num_layouts = g_strv_length (str_layouts);
+
+    if (num_layouts > NNS_TENSOR_SIZE_LIMIT) {
+      GST_WARNING ("Invalid param, layouts (%d) max (%d)\n",
+          num_layouts, NNS_TENSOR_SIZE_LIMIT);
+
+      num_layouts = NNS_TENSOR_SIZE_LIMIT;
+    }
+
+    for (i = 0; i < num_layouts; i++) {
+      layout[i] = gst_tensor_parse_layout_string (str_layouts[i]);
+    }
+
+    g_strfreev (str_layouts);
+  }
+
+  return num_layouts;
+}
+
+/**
+ * @brief Get layout string of tensor layout.
+ * @param layout layout of the tensor
+ * @return string of layout in tensor
+ * @note The returned value should be freed with g_free()
```

Comment:
plz remove this line (no ned to free string)

Reply:
updated.